WhatsNew/Items/WhatsNew.swift

```diff
@@ -9,31 +9,38 @@ import Foundation
 
 public struct WhatsNew {
     static let bundle = Bundle(for: WhatsNewViewController.self)
-    static let appVersion = Bundle.main.infoDictionary?["CFBundleShortVersionString"] as? String
-    static let userDefaultsKey = "LatestAppVersionPresented"
+    public static let appVersion = Bundle.main.infoDictionary?["CFBundleShortVersionString"] as? String
+    static let userDefaultsKeyLatestAppVersionPresented = "LatestAppVersionPresented"
 
     static func markCurrentVersionAsPresented() {
-        UserDefaults.standard.set(appVersion, forKey: userDefaultsKey)
-        UserDefaults.standard.synchronize()
+        UserDefaults.standard.set(appVersion, forKey: userDefaultsKeyLatestAppVersionPresented)
     }
 
-    public static func shouldPresent(with option: PresentationOption = .always) -> Bool {
-        let previousAppVersion = UserDefaults.standard.string(forKey: userDefaultsKey)
+    public static func shouldPresent(with option: PresentationOption = .always, currentVersion: String? = appVersion) -> Bool {
+        guard let currentAppVersion = currentVersion else { return false }
+        let previousAppVersion = UserDefaults.standard.string(forKey: userDefaultsKeyLatestAppVersionPresented)
+        let didUpdate = previousAppVersion != currentAppVersion
         
-        let didUpdate = previousAppVersion != appVersion
-        
-        // Choose based on the selected presentation option.
         switch option {
         case .debug: return true
         case .never: return false
-        case .majorVersion: return didChangeMajorVersion(previous: previousAppVersion, current: appVersion)
+        case .majorVersion: return didUpdate && isMajorVersion(version: currentAppVersion)
         case .always: return didUpdate
         }
     }
-
-    private static func didChangeMajorVersion(previous: String?, current: String?) -> Bool {
```

Comment:
I don't see the problem with my implementation of `didChangeMajorVersion` in this case?

Reply:
For `previous = nil` your implementation did not present the whats new controller when presentation option was set to `.major`.

Reply:
This is why I changed the implementation to a better understandable way: I check for `didUpdate && isMajorVersion`.